deepmd/utils/data.py

```diff
@@ -249,6 +254,25 @@ def check_test_size(self, test_size):
         else:
             return None
 
+    def get_item(self, index: int) -> dict:
+        """Get a single frame data . The frame is picked from the data system by index.
+
+        Parameters
+        ----------
+        index
+            index of the frame
+        """
+        for i in range(
+            0, len(self.dirs)
+        ):  # note: if different sets can be merged, prefix sum is unused to calculate
+            if index < self.prefix_sum[i]:
+                break
```

Comment:
I guess you need [bisect](https://docs.python.org/3/library/bisect.html). This can be simply written as
```suggestion
        i = bisect.bisect_right(self.prefix_sum, index)
```